scripts/build.sh

```diff
@@ -10,78 +10,36 @@ if [[ "x$version" == "x" ]]; then
   exit 1
 fi
 
-# Create a direcotry to save assets
-ASSETSDIR=release
-mkdir $ASSETSDIR
-
-function build_amd64_linux() {
-  DIR=pulsarctl-amd64-linux
-  mkdir ${DIR}
-  CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o pulsarctl -ldflags "-X github.com/streamnative/pulsarctl/pkg/pulsar.ReleaseVersion=Pulsarctl-Go-$version" .
-  mv pulsarctl ${DIR}
-  cp -r plugins ${DIR}
-  tar -czf ${DIR}.tar.gz ${DIR}
-  mv ${DIR}.tar.gz $ASSETSDIR
-}
+ROOT_DIR=`cd $(dirname $0)/../; pwd`
 
-function build_386_linux() {
-  DIR=pulsarctl-386-linux
-  mkdir ${DIR}
-  CGO_ENABLED=0 GOOS=linux GOARCH=386 go build -o pulsarctl -ldflags "-X github.com/streamnative/pulsarctl/pkg/pulsar.ReleaseVersion=Pulsarctl-Go-$version" .
-  mv pulsarctl ${DIR}
-  cp -r plugins ${DIR}
-  tar -czf ${DIR}.tar.gz ${DIR}
-  mv ${DIR}.tar.gz $ASSETSDIR
-}
+pushd $ROOT_DIR
 
-function build_amd64_darwin() {
-  DIR=pulsarctl-amd64-darwin
-  mkdir ${DIR}
-  CGO_ENABLED=0 GOOS=darwin GOARCH=amd64 go build -o pulsarctl -ldflags "-X github.com/streamnative/pulsarctl/pkg/pulsar.ReleaseVersion=Pulsarctl-Go-$version" .
-  mv pulsarctl ${DIR}
-  cp -r plugins ${DIR}
-  tar -czf ${DIR}.tar.gz ${DIR}
-  mv ${DIR}.tar.gz $ASSETSDIR
-}
-
-function build_arm64_darwin() {
-  DIR=pulsarctl-arm64-darwin
-  mkdir ${DIR}
-  CGO_ENABLED=0 GOOS=darwin GOARCH=arm64 go build -o pulsarctl -ldflags "-X github.com/streamnative/pulsarctl/pkg/pulsar.ReleaseVersion=Pulsarctl-Go-$version" .
-  mv pulsarctl ${DIR}
-  cp -r plugins ${DIR}
-  tar -czf ${DIR}.tar.gz ${DIR}
-  mv ${DIR}.tar.gz $ASSETSDIR
-}
+LDFLAGS+="-X \"github.com/streamnative/pulsarctl/pkg/cmdutils.ReleaseVersion=${version}\" "
+LDFLAGS+="-X \"github.com/streamnative/pulsarctl/pkg/cmdutils.BuildTS=$(date -u '+%Y-%m-%d %H:%M:%S')\" "
+LDFLAGS+="-X \"github.com/streamnative/pulsarctl/pkg/cmdutils.GitHash=$(git rev-parse HEAD)\" "
+LDFLAGS+="-X \"github.com/streamnative/pulsarctl/pkg/cmdutils.GitBranch=$(git rev-parse --abbrev-ref HEAD)\" "
+LDFLAGS+="-X \"github.com/streamnative/pulsarctl/pkg/cmdutils.GoVersion=$(go version)\" "
```

Comment:
It will be duplicate to set LDFLAGS in `build.sh` again since Makefile has already set LDFLAGS. If we want to add/modify a field in version print later, we need to modify the `build.sh` and Makefile together.
I prefer to move the build logic from `build.sh` to Makefile, and use the make command in `build.sh` to triggle.

Take  `build_amd64_linux` for example:

In the Makefile, we can add

```
amd64_linux:
     CGO_ENABLED=0 GOOS=linux GOARCH=amd64 $(GOBUILD) -ldflags '$(LDFLAGS)' -o bin/pulsarctl
```

In the `build.sh`, it will be

```
function build_amd64_linux() {
  DIR=pulsarctl-amd64-linux
  mkdir ${DIR}
  make amd64_linux
  mv pulsarctl ${DIR}
  cp -r plugins ${DIR}
  tar -czf ${DIR}.tar.gz ${DIR}
  mv ${DIR}.tar.gz $ASSETSDIR
}

```